app/models/assignment.rb

```diff
@@ -17,4 +17,20 @@ def copy
     result.tests = tests.map(&:dup)
     result
   end
+
+  def submission_for(student)
+    if submissions.find { |s| s.uploaded_by_id == student.user_id }
+      submissions
+    end
+  end
+
+  def submission_status_for(student)
```

Comment:
I would move this to the decorator, which is the "glue" between the model and the UI. You can put view-related methods there. For example, you can create a method

`submission_status_for(student)` in the AssignmentDecorator, and call it in the view using `assignment.decorate.submission_status_for(student)`. That will also be the right place for applying layout, such as colors.